pkg/controller/vsphere/session/session.go

```diff
@@ -156,3 +165,20 @@ func (s *Session) GetTask(ctx context.Context, taskRef string) (*mo.Task, error)
 	}
 	return &obj, nil
 }
+
+func (s *Session) WithRestClient(ctx context.Context, f func(c *rest.Client) error) error {
```

Comment:
the passed parameter `func(c *rest.Client)` does not seem to be used?

Reply:
nvm, I see is required to ensure the `f(c)` call below.
Do we need the login/logout calls?
What's the reason why we can not simplify by just passing `rest.NewClient(s.Client.Client)` where needed with out this wrapper? e.g `vm.getRegionAndZone(rest.NewClient(s.Client.Client), regionLabel, zoneLabel)`

Reply:
The function returns the result of applying `f()`: `return f(c)`

Reply:
> Do we need the login/logout calls?

Yes, or else the REST client is not authenticated.

> What's the reason why we can not simplify by just passing rest.NewClient(s.Client.Client) where needed with out this wrapper? e.g vm.getRegionAndZone(rest.NewClient(s.Client.Client), regionLabel, zoneLabel)

We could pass a rest client directly to that method, but we'd have to authenticate it. Doing that in any place that might need it doesn't seem like a simplification.